MacOS CI fails

```
nox > python -m pip install --constraint=.nox/tests-3-8/tmp/requirements.txt 'coverage[toml]' pytest pygments
nox > coverage run --parallel -m pytest
============================= test session starts ==============================
platform darwin -- Python 3.8.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/runner/work/ehrapy/ehrapy, configfile: pyproject.toml
collected 163 items
tests/anndata/test_anndata_ext.py ......................                 [ 13%]
tests/core/test_str_matching.py .....                                    [ 16%]
tests/io/test_read.py ..........................                         [ 32%]
tests/preprocessing/test_encode.py ...............                       [ 41%]
tests/preprocessing/test_highly_variable_features.py .                   [ 42%]
tests/preprocessing/test_imputation.py ................................. [ 62%]
........Fatal Python error: Segmentation fault
Thread 0x000070000542e000 (most recent call first):
  File "/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/dataclasses.py", line 1283 in replace
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/rich/table.py", line 116 in copy
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/rich/progress.py", line 1554 in <genexpr>
Thread 0x000000011a5b1e00 (most recent call first):
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/lightgbm/basic.py", line 1659 in __init_from_np2d
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/lightgbm/basic.py", line 1538 in _lazy_init
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/lightgbm/basic.py", line 1815 in construct
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/lightgbm/basic.py", line 2605 in __init__
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/lightgbm/engine.py", line 271 in train
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/miceforest/ImputationKernel.py", line 919 in mice
  File "/Users/runner/work/ehrapy/ehrapy/ehrapy/preprocessing/_data_imputation.py", line 1010 in _miceforest_impute
  File "/Users/runner/work/ehrapy/ehrapy/ehrapy/preprocessing/_data_imputation.py", line 966 in mice_forest_impute
  File "/Users/runner/work/ehrapy/ehrapy/tests/preprocessing/test_imputation.py", line 289 in test_miceforest_impute_no_copy
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/python.py", line 1761 in runtest
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 338 in from_call
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 219 in call_and_report
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/main.py", line 347 in pytest_runtestloop
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/main.py", line 322 in _main
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/main.py", line 268 in wrap_session
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/main.py", line 315 in pytest_cmdline_main
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/config/__init__.py", line 164 in main
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/_pytest/config/__init__.py", line 187 in console_main
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/pytest/__main__.py", line 5 in <module>
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/coverage/execfile.py", line 199 in run
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/coverage/cmdline.py", line 830 in do_run
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/coverage/cmdline.py", line 659 in command_line
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/lib/python3.8/site-packages/coverage/cmdline.py", line 943 in main
nox > Command coverage run --parallel -m pytest failed with exit code -11
  File "/Users/runner/work/ehrapy/ehrapy/.nox/tests-3-8/bin/coverage", line 8 in <module>
nox > Session tests-3.8 failed.
Error: Process completed with exit code 1.
```

A segfault? Really not sure about what's going on.

More details by @namsaraeva who can reproduce this locally:
```
FAILED [ 85%]Quality control metrics missing. Calculating...
Running miceforest ⠋

tests/preprocessing/test_imputation.py:286 (TestImputation.test_miceforest_impute_no_copy)
self = <tests.preprocessing.test_imputation.TestImputation object at 0x7ff3f4d55af0>

    def test_miceforest_impute_no_copy(self):
        adata = read_csv(dataset_path=f"{_TEST_PATH}/test_impute_iris.csv")
>       adata_imputed = mice_forest_impute(adata)

test_imputation.py:289: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../ehrapy/preprocessing/_data_imputation.py:1059: in mice_forest_impute
    _miceforest_impute(
../../ehrapy/preprocessing/_data_imputation.py:1089: in _miceforest_impute
    import miceforest as mf
../../../../miniconda3/envs/ehrapy/lib/python3.9/site-packages/miceforest/__init__.py:14: in <module>
    from .ImputationKernel import ImputationKernel
../../../../miniconda3/envs/ehrapy/lib/python3.9/site-packages/miceforest/ImputationKernel.py:23: in <module>
    from lightgbm import train, Dataset, cv, log_evaluation, early_stopping, Booster
../../../../miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/__init__.py:8: in <module>
    from .basic import Booster, Dataset, Sequence, register_logger
../../../../miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/basic.py:110: in <module>
    _LIB = _load_lib()
../../../../miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/basic.py:101: in _load_lib
    lib = ctypes.cdll.LoadLibrary(lib_path[0])
../../../../miniconda3/envs/ehrapy/lib/python3.9/ctypes/__init__.py:460: in LoadLibrary
    return self._dlltype(name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <CDLL '/Users/thenamsa/miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/lib_lightgbm.so', handle 0 at 0x7ff3f55e58e0>
name = '/Users/thenamsa/miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/lib_lightgbm.so'
mode = 4, handle = None, use_errno = False, use_last_error = False
winmode = None

    def __init__(self, name, mode=DEFAULT_MODE, handle=None,
                 use_errno=False,
                 use_last_error=False,
                 winmode=None):
        self._name = name
        flags = self._func_flags_
        if use_errno:
            flags |= _FUNCFLAG_USE_ERRNO
        if use_last_error:
            flags |= _FUNCFLAG_USE_LASTERROR
        if _sys.platform.startswith("aix"):
            """When the name contains ".a(" and ends with ")",
               e.g., "libFOO.a(libFOO.so)" - this is taken to be an
               archive(member) syntax for dlopen(), and the mode is adjusted.
               Otherwise, name is presented to dlopen() as a file argument.
            """
            if name and name.endswith(")") and ".a(" in name:
                mode |= ( _os.RTLD_MEMBER | _os.RTLD_NOW )
        if _os.name == "nt":
            if winmode is not None:
                mode = winmode
            else:
                import nt
                mode = nt._LOAD_LIBRARY_SEARCH_DEFAULT_DIRS
                if '/' in name or '\\' in name:
                    self._name = nt._getfullpathname(self._name)
                    mode |= nt._LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR
                # PATH is how DLL lookup has always worked in the past
                # in Python on Windows. IMHO both the above mode flags
                # are not wanted and cause many serious regressions within
                # the conda ecosystem on Windows. We should however
                # propagate any PATH changes that have happened to Python
                # library and that is not yet implemented.
                LOAD_WITH_ALTERED_SEARCH_PATH = 0x00000008
                mode = LOAD_WITH_ALTERED_SEARCH_PATH
    
        class _FuncPtr(_CFuncPtr):
            _flags_ = flags
            _restype_ = self._func_restype_
        self._FuncPtr = _FuncPtr
    
        if handle is None:
>           self._handle = _dlopen(self._name, mode)
E           OSError: dlopen(/Users/thenamsa/miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/lib_lightgbm.so, 0x0006): Library not loaded: /usr/local/opt/libomp/lib/libomp.dylib
E             Referenced from: /Users/thenamsa/miniconda3/envs/ehrapy/lib/python3.9/site-packages/lightgbm/lib_lightgbm.so
E             Reason: tried: '/usr/local/opt/libomp/lib/libomp.dylib' (no such file), '/usr/local/lib/libomp.dylib' (no such file), '/usr/lib/libomp.dylib' (no such file)

../../../../miniconda3/envs/ehrapy/lib/python3.9/ctypes/__init__.py:382: OSError
```

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -32,11 +32,11 @@ jobs:
                           os: ubuntu-latest,
                           session: "tests",
                       }
-                            - {
-                                  python-version: 3.8,
-                                  os: macos-latest,
-                                  session: "tests",
-                              }
+                    - {
+                          python-version: 3.8,
+                          os: macos-latest,
+                          session: "tests",
+                      }
         #                     - {
         #                           python-version: 3.8,
         #                           os: ubuntu-latest,

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -32,11 +32,11 @@ jobs:
                           os: ubuntu-latest,
                           session: "tests",
                       }
-        #                    - {
-        #                          python-version: 3.8,
-        #                          os: macos-latest,
-        #                          session: "tests",
-        #                      }
+                            - {
+                                  python-version: 3.8,
+                                  os: macos-latest,
+                                  session: "tests",
+                              }
         #                     - {
         #                           python-version: 3.8,
         #                           os: ubuntu-latest,

